File: java/3_operator/bitwise/ShiftLeft.java

class ShiftLeft {
    
    public static void main(String[] args){
        int a = 3; String bitA;
        int hasil = a << 3;
        String bitHasil;
        
        bitA = String.format("%8s", Integer.toBinaryString(a)).replace(' ', '0');
        bitHasil = String.format("%8s", Integer.toBinaryString(hasil)).replace(' ','0');

        System.out.println("Hasil bitwise << sebanyak 3 variable a adalah:");
        System.out.println(bitA);
        System.out.println(bitHasil);
    }   
}

File: kotlin/3_operator/bitwise/ShiftLeft.kt

fun main() {

    val a: Int = 3
    var hasil = a shl 1

    val bitA = a.toString(2).padStart(8,'0')
    val bitHasil = hasil.toString(2).padStart(8,'0')

    println("Hasil bitwise Shift left 1kali variable a adalah:")
    println(bitA)
    println(bitHasil)
}
